stl/inc/atomic

```diff
@@ -857,7 +857,13 @@ struct _Atomic_integral; // not defined
 template <class _Ty>
 struct _Atomic_integral<_Ty, 1> : _Atomic_storage<_Ty> { // atomic integral operations using 1-byte intrinsics
     using _Base = _Atomic_storage<_Ty>;
+
+#ifdef __cplusplus_winrt // TRANSITION, VSO-1083296
+    _Atomic_integral() = default;
+    /* implicit */ constexpr _Atomic_integral(const _Ty _Value) noexcept : _Base(_Value) {}
+#else // ^^^ workaround / no workaround vvv
     using _Base::_Base;
+#endif // ^^^ no workaround ^^^

```

Comment:
I'm a little torn between being OK with this and wanting to just unconditionally do the workaround. I get that we want to change it back, but I don't like ifdefs and we can still mark it with a transition.

Reply:
I'm certainly not a fan of preprocessor logic, but I chose this solution because at this time, the compiler bug is thought to affect C++/CX only - not C++/CLI and not native compilation. Therefore, I would prefer to limit the potential disruption of the workaround, since we have more experience with the existing code using inheriting constructors.